### PR TITLE
Make more List and Results tests work

### PR DESCRIPTION
**Pull request by jedelbo**
The tests not yet passing are commented out. They should - however - be able to compile.

### CODE REVIEW

**Comment by jedelbo** [comment]
@tgoyne Perhaps this would be a good time to review this PR. I have made some changes to the implementation that I would like you to evaluate.

File: src/results.cpp [hunk 1]
```diff
@@ -47,47 +47,51 @@ Results::Results(SharedRealm r, Table& table)
...
 
 bool Results::is_valid() const
 {
     if (m_realm) {
         m_realm->verify_thread();
     }
 
-//    if (m_table && !m_table->is_valid())
-//        return false;
+    if (((const Table*)m_table) != nullptr)
```
Comment (tgoyne) [inline src/results.cpp]: This is an awfully longwinded way to write `if (m_table)`

File: src/results.cpp [hunk 2]
```diff
@@ -47,47 +47,51 @@ Results::Results(SharedRealm r, Table& table)
...
+        return (bool)m_table;
```
Comment (tgoyne) [inline src/results.cpp]: This codebase needs to be `-Wold-style-cast` clean.

File: src/results.cpp [hunk 3]
```diff
@@ -230,11 +242,12 @@ void Results::evaluate_query_if_needed(bool wants_notifications)
         return;
     }
 
-    if (m_notifier)
+    // if (m_notifier) FIXME: Why was this condition added?
```
Comment (tgoyne) [inline src/results.cpp]: This looks like it was just a stray paste and should be deleted.

File: src/results.cpp [hunk 4]
```diff
@@ -488,6 +503,9 @@ Query Results::get_query() const
             if (query.get_table()) {
                 return query;
             }
+            if (m_table_view.is_backlink_view()) {
+                return Query();
```
Comment (tgoyne) [inline src/results.cpp]: This does need to return a real query as we support filtering and sorting backlink views.

File: src/results.hpp [hunk 5]
```diff
@@ -295,6 +301,28 @@ size_t Results::index_of(Context& ctx, T value)
 {
     return dispatch([&](auto t) { return this->index_of(ctx.template unbox<std::decay_t<decltype(*t)>>(value)); });
 }
+
+template<typename T>
+auto Results::list_get_as(size_t ndx) const
```
Comment (tgoyne) [inline src/results.hpp]: There doesn't appear to be any reason for these to be defined in the header.

File: tests/list.cpp [hunk 6]
```diff
@@ -425,6 +425,8 @@ TEST_CASE("list") {
             REQUIRE_MOVES(change, {1, 2});
         }
 
+#if 0
+        // FIXME irrelevant?
```
Comment (tgoyne) [inline tests/list.cpp]: Yes, everything related to moving rows in tables should just be deleted.

Comment (jedelbo) [reply tests/list.cpp]: Deleted

File: src/results.hpp (hunk 5 above)
Comment (jedelbo) [reply src/results.hpp]: Moved to .cpp

File: src/results.cpp (hunk 3 above)
Comment (jedelbo) [reply src/results.cpp]: Done

File: src/results.cpp (hunk 2 above)
Comment (jedelbo) [reply src/results.cpp]: Fixed

File: src/results.cpp (hunk 1 above)
Comment (jedelbo) [reply src/results.cpp]: Explanation added

File: src/results.cpp (hunk 4 above)
Comment (jedelbo) [reply src/results.cpp]: I realize that this was just a cover for underlying problems. Made some fixes in core (setting `m_view` when importing) and results_notifier.cpp (setting of `m_results_were_used`)

File: src/list.cpp [hunk 7]
```diff
@@ -111,10 +127,15 @@ static StringData object_name(Table const& table)
 ObjectSchema const& List::get_object_schema() const
 {
     verify_attached();
-    REALM_ASSERT(get_type() == PropertyType::Object);
```
[the remark this answers is not in the document]
Comment (jedelbo) [reply src/list.cpp]: Reverted

**Pull request merged**